internal/gitutil/git_test.go

```diff
@@ -0,0 +1,213 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitutil
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"sigs.k8s.io/krew/internal/testutil"
+)
+
+func assertRepoIsCloned(t *testing.T, repoPath string) {
+	t.Helper()
+
+	ok, err := IsGitCloned(repoPath)
+
+	if err != nil {
+		t.Errorf("expected to finish correctly: %s", err)
+	}
+
+	if ok != true {
+		t.Errorf("expected to return true after cloning the repo")
+	}
+}
+
+func TestEnsureClonedClonesARepository(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	httpClonePath := tempDir.Path("krew-from-https")
+	localClonePath := tempDir.Path("krew-from-local-path")
+
+	if err := EnsureCloned("https://github.com/kubernetes-sigs/krew.git", httpClonePath); err != nil {
+		t.Errorf("http clone expected to finish correctly: %s", err)
+	}
+
+	if err := EnsureCloned(httpClonePath, localClonePath); err != nil {
+		t.Errorf("folder clone expected to finish correctly: %s", err)
+	}
+
+	assertRepoIsCloned(t, httpClonePath)
+	assertRepoIsCloned(t, localClonePath)
+}
+
+func TestEnsureClonedDoesntCloneAnExistingRepo(t *testing.T) {
+	tempDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	if err := EnsureCloned("https://github.com/kubernetes-sigs/krew.git", tempDir.Root()); err != nil {
+		t.Errorf("clone expected to finish correctly: %s", err)
+	}
+
+	tempDir.Write("file", []byte("create a file to ensure is not overwriten in the next clone"))
+
+	if err := EnsureCloned("https://github.com/kubernetes-sigs/krew.git", tempDir.Root()); err != nil {
```

Comment:
```suggestion
	if err := EnsureCloned("https://github.com/kubernetes-sigs/krew.git", tempDir.Root()); err == nil {
```
I would expect that the second invocation provokes an error. Are you sure this is correct?

Reply:
i'll double check but I think it just returned without an error.